File: public/code/drafting-emails-via-the-api/2.py
import requests

# You can find this key in your API requests page:
# https://buttondown.email/requests
BUTTONDOWN_API_KEY = "your-api-key-here"

EMAIL_ID = "your-email-id-here"

headers = {"Authorization": f"Token #{BUTTONDOWN_API_KEY}"}
BASE_URL = "https://api.buttondown.email"
ENDPOINT = f"/v1/emails/{EMAIL_ID}"
response = requests.get(f"{BASE_URL}{ENDPOINT}", headers=headers)
print(response.json())

File: public/code/drafting-emails-via-the-api/2.rb
require "HTTParty"
require "json"

# You can find this key in your API requests page:
# https://buttondown.com/requests
BUTTONDOWN_API_KEY = "your-api-key-here"

response = HTTParty.get(
  "https://api.buttondown.email/v1/emails/#{EMAIL_ID}",
  headers: {
    "Authorization" => "Token #{BUTTONDOWN_API_KEY}",
    "Content-Type" => "application/json"
  }
)
JSON.parse(response.body)
print(response)
